src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -211,18 +211,27 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 			bool updated = false;
 			HashSet<string> files = new HashSet<string> ();
 			foreach (var entry in zip) {
+				progressCallback?.Invoke (i++, total);
 				if (entry.FullName.Contains ("/__MACOSX/") ||
 						entry.FullName.EndsWith ("/__MACOSX", StringComparison.OrdinalIgnoreCase) ||
 						entry.FullName.EndsWith ("/.DS_Store", StringComparison.OrdinalIgnoreCase))
 					continue;
 				var fullName = modifyCallback?.Invoke (entry.FullName) ?? entry.FullName;
 				if (entry.IsDirectory) {
-					Directory.CreateDirectory (Path.Combine (destination, fullName));
+					try {
+						Directory.CreateDirectory (Path.Combine (destination, fullName));
+					} catch (NotSupportedException ex) {
+						//NOTE: invalid paths, such as `:` on Windows can cause this
+						throw new NotSupportedException ($"Invalid zip entry `{fullName}` found in archive.", ex);
+					}
```

Comment:
### TODO

Need to investigate two options here:
- What happens if we fix the path to replace `:` with `-`? Is the error condition any better?
- If we are going to `throw` here, we need to make sure an XA error code is logged, which isn't happening in this PR currently.

Reply:
@jonpryor so if I try No. 1 here and replace `:` with `-`, then the build succeeds like it does on macOS.

However, we are likely going to have invalid resource IDs at runtime, since none of the referenced assembly's IDs are getting set, `UpdateIdValues` is blank in my test project:
```csharp
public static void UpdateIdValues()
{
}
```
Makes sense, because `aapt` (or our MSBuild tasks) have no way to find the extracted resources that exist in the broken `library_project_importsC-/projects/pclcrypto/src/.obj/pclcrypto/Release/monoandroid23/jl/res` directory.

I think we should keep this PR as-is, but add an XA error code. I'll also make sure `PathTooLongException` gets the same error code.